controllers/dependencies.go

```diff
@@ -72,34 +75,67 @@ func (r *OperatorPipelineReconciler) reconcilePipelineDependencies(meta metav1.O
 			return err
 		}
 	}
+
 	return nil
 }
 
-func (r *OperatorPipelineReconciler) removePipelineDependencyFiles(filePath string) error {
-	if err := os.RemoveAll(filePath); err != nil {
-		log.Error(err, "Couldn't remove operator-pipelines directory")
+func (r *OperatorPipelineReconciler) applyPipelineManifests(fileName string, meta metav1.ObjectMeta, obj *tekton.Pipeline) error {
+
+	b, err := os.ReadFile(fileName)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Couldn't read manifest file for: %s", fileName))
+		return err
+	}
+
+	if err = yamlutil.Unmarshal(b, &obj); err != nil {
+		log.Error(err, "Couldn't unmarshall yaml file")
 		return err
 	}
+
+	obj.SetNamespace(meta.Namespace)
+	err = r.Get(context.Background(), types.NamespacedName{Name: obj.Name, Namespace: obj.Namespace}, obj)
+	if err != nil && errors.IsNotFound(err) {
+
+		if err = r.Client.Create(context.Background(), obj); err != nil {
+			log.Error(err, "failed to create pipeline resource")
+			return err
+		}
+	} else {
+		if err = r.Client.Update(context.Background(), obj); err != nil {
+			log.Error(err, "failed to update pipeline resource")
+			return err
```

Comment:
I feel like we don't handle the case where error is anything but IsNotFound. It will jump to the else block to update the resource, which should be avoided IMO. I think this should be the code structure:
```
err = r.Get(context.Background(), types.NamespacedName{Name: obj.Name, Namespace: obj.Namespace}, obj)

if (obj != nil) {
   // update the object and handle error
}
if (err != nil){
   if errors.IsNotFound(err) {
     // create resource
   } else { //this is error
        return err
  }
}
return nil
```

Reply:
You could even get rid of the else altogether. 

if err != nil {
  if !errors.IsNotFound(err) {
    return err
  }
  createResource()
}
return nil

Reply:
I can't check for `obj !=nil` since it is already pre-populated when marshaled earlier. I'll check to see if `obj` has a `UUID` if it does it means it came from the cluster.

Reply:
Took both of your suggests, please look at latest code.